src/AccountManager.ts

```diff
@@ -0,0 +1,128 @@
+import sigUtil, {
+  SignTypedDataVersion,
+  TypedMessage,
+} from '@metamask/eth-sig-util';
+import type { EnvelopingTypes } from '@rsksmart/rif-relay-contracts/dist/typechain-types/contracts/RelayHub';
+import { providers, Wallet } from 'ethers';
+import { getAddress } from 'ethers/lib/utils';
+import {
+  deployRequestType,
+  EnvelopingMessageTypes,
+  getEnvelopingRequestDataV4Field,
+  relayRequestType,
+} from './typedRequestData.utils';
+
+export type RelayOrDeployRequest =
+  | EnvelopingTypes.RelayRequestStruct
+  | EnvelopingTypes.DeployRequestStruct;
+
+export default class AccountManager {
+  private _provider: providers.Provider;
+
+  private _accounts: Wallet[] = [];
+
+  chainId: number;
+
+  constructor(provider: providers.Provider, chainId: number) {
+    this._provider = provider;
+    this.chainId = chainId;
+  }
+
+  getAccounts(): string[] {
+    return this._accounts.map((it) => it.address);
+  }
+
+  addAccount(account: Wallet): void {
+    const wallet = new Wallet(account.privateKey, this._provider);
+    if (wallet.address !== account.address) {
+      throw new Error('invalid keypair');
+    }
+    this._accounts.push(wallet);
+  }
+
+  isDeployRequest({ request }: RelayOrDeployRequest): boolean {
+    return 'index' in request;
+  }
+
+  async sign(relayRequest: RelayOrDeployRequest): Promise<string> {
+    const {
+      relayData: { callForwarder },
+      request: { from },
+    } = relayRequest as typeof relayRequest & {
+      relayData: { callForwarder: string };
+      request: { from: string };
+    }; // this is here just to remove PromiseOrValue<string> type from the def string properties
```

Comment:
because then we'd always have to either cast or convert the variable to string. The type exists for their (typechain) ethers v5 target to avoid multiple awaits in nested calls. This is not our case however.
But yea, I guess this casting may cause issue if in future this type changes. I'll just convert then.

Reply:
done